supervision/dataset/utils.py

```diff
@@ -46,6 +47,59 @@ def approximate_mask_with_polygons(
     ]
 
 
+def merge_polygons(
```

Comment:
I think the `merge_polygons` name is a bit misleading and not conststant with outhe util names in supervision. I think we should call it `polygons_to_mask`.

And of course we already have a function called `polygon_to_mask` that does the same but only take one mask. 

It seems a bit silly that we would have 2 functions doeng almost the same. I thin kwe should:
- rename `merge_polygons` to `polygons_to_mask`.
- make sure that `polygons_to_mask` is not using `polygon_to_mask` internally.
- modify places where `polygon_to_mask` was used and use `polygons_to_mask` instead.
- mask `polygon_to_mask` as deprecated and recommend using `polygons_to_mask` instead.

Reply:
@DancinParrot, here is an example of how we mark functions as deprecated. We use `deprecated` decorator to do it.

```
@deprecated(
    "`Detections.from_roboflow` is deprecated and will be removed in "
    "`supervision-0.22.0`. Use `Detections.from_inference` instead."
)
```

Reply:
Please use `supervision-0.27.0` as a deprecation version.